test-utils/testlib/src/alphanet_utils.rs

```diff
@@ -453,8 +501,23 @@ pub fn sample_two_nodes(num_nodes: usize) -> (usize, usize) {
 pub fn sample_queryable_node(nodes: &Vec<Box<Node>>) -> usize {
     let num_nodes = nodes.len();
     let mut k = rand::random::<usize>() % num_nodes;
-    while nodes[k].node_type() == NodeType::ProcessNode {
+    while !nodes[k].is_running() {
         k = rand::random::<usize>() % num_nodes;
     }
     k
-}
\ No newline at end of file
+}
+
+pub fn wait_for_catchup(nodes: &Vec<Box<Node>>) {
```

Comment:
This might never happen with large number of nodes even if they have perfect conditions.

Reply:
I think it only makes sense to have three types of wait-checks:
1. Wait until sufficient number of nodes is caught up (> 2/3). This can be checked by looking at the block indices and verifying that the blocks are produced;
2. Wait until a certain node is caught up and participating in a consensus. Check first-layer BLS signatures;
3. Wait until all nodes are more-or-less caught up. Check that the max_block_index - min_block_index < threshold;